common/workunit/workunit.cpp

```diff
@@ -9492,9 +9492,25 @@ IPropertyTree *CLocalWorkUnit::getUnpackedTree(bool includeProgress) const
     return ret.getClear();
 }
 
+IPropertyTree *CLocalWorkUnit::queryIndirectTree() const
+{
+    if (indirectTree)
+        return indirectTree;
+    if (!p->hasProp("@clonedFromWorkunit"))
+        return p;
+    else
+    {
+        indirectTree.setown(createPTreeFromIPT(p));
+        Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
```

Comment:
can this fail if not connected to dali (e.g. disconnected during query)

Reply:
Few other calls to getWorkUnitFactory in this module check it, so I'm guessing not (or it it is a problem, it's not specific to this new code)